web/server/entrypoint.go

```diff
@@ -479,7 +502,8 @@ func (s *robotServer) serveWeb(ctx context.Context, cfg *config.Config) (err err
 	}()
 	defer cancel()
 
-	options, err := s.createWebOptions(processedConfig)
+	// Create initial web options with `minimalProcessedConfig`.
+	options, err := s.createWebOptions(minimalProcessedConfig)
```

Comment:
I can't comment off-diff. The goroutine spun off above will check for `diff.NetworkEqual` and if not, run `myRobot.StartWeb` (newline 490).

Just below this we call `web.RunWeb`. I'm not sure what the significance is between calling `StartWeb` and `RunWeb`, but assuming that's not interesting: is it possible for those two things to race? Are we guaranteed to end up with the right set of web`options`?

To clarify, this is a question about existing behavior, nothing this patch is necessarily touching.

Reply:
> The goroutine spun off above will check for diff.NetworkEqual and if not, run myRobot.StartWeb (newline 490).

Correct; I believe there's a call to `StopWeb` before that happens, too. And then a `Reconfigure` after that `StopWeb`. All about handling network changes in the config.

> Just below this we call web.RunWeb. I'm not sure what the significance is between having different methods, StartWeb and RunWeb, but assuming that's not interesting

It's interesting having those two methods. `StartWeb` starts up the web service on the robot. `RunWeb` does that, but also waits on `<-ctx.Done()`, so it's a blocking call and represents the "main" program that "runs" when you call `go run web/cmd/server/main.go`. 

> Is it possible for those two things to race? Are we guaranteed to end up with the right set of weboptions?

It depends what you mean by race. There is a lock on starting the web service, so I'm not sure we'd see a race manifest as an actual `DATA RACE`, but I think you are "right" in wondering about the "right set of weboptions." I'm not _entirely_ sure, but I think `RunWeb` would run into an error if it tried to start the web service with an old set of options after the config watcher goroutine had started it already with a new set of options. So, my _guess_ is we'd see an error from `RunWeb` and an inability to start the server in the event of the race you're describing.